clippy_lints/src/manual_clamp.rs

```diff
@@ -103,6 +105,25 @@ struct ClampSuggestion<'tcx> {
     hir_with_ignore_attr: Option<HirId>,
 }
 
+impl<'tcx> ClampSuggestion<'tcx> {
+    /// This function will return true if and only if you can demonstrate at compile time that min
+    /// is less than max.
+    fn min_less_than_max(&self, cx: &LateContext<'tcx>) -> bool {
+        let max_type = cx.typeck_results().expr_ty(self.params.max);
+        let min_type = cx.typeck_results().expr_ty(self.params.min);
+        if max_type != min_type {
+            return false;
+        }
+        if let Some(max) = constant(cx, cx.typeck_results(), self.params.max)
+            && let Some(min) = constant(cx, cx.typeck_results(), self.params.min)
+        {
+            Constant::partial_cmp(cx.tcx, max_type, &min, &max).is_some_and(|o| o != Ordering::Greater)
```

Comment:
We could push it even further. :smile_cat: 

```suggestion
        if let Some(max) = constant(cx, cx.typeck_results(), self.params.max)
            && let Some(min) = constant(cx, cx.typeck_results(), self.params.min)
            && let Some(ord) = Constant::partial_cmp(cx.tcx, max_type, &min, &max)
        {
            ord != Ordering::Greater
```

Reply:
Nice! Resolved in 4d7d66462